zbuf/merger.go

```diff
@@ -186,3 +188,13 @@ func (m *Merger) Pull() (Batch, error) {
 func (m *Merger) Cancel() {
 	m.cancel()
 }
+
+func deterministicCompare(fn expr.CompareFn) expr.CompareFn {
```

Comment:
Nits: Put this after NewCompareFn. And deterministicCompare doesn't really describe what this does (it isn't any more or less deterministic than fn), so maybe call it totalOrderCompare or something else that expresses the intent, which I'd say is to describe a total order base on fn.